python/DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,68 @@
+# cuML python developer guide
+This document summarizes rules and best practices for contributions to the cuML python component of rapidsai/cuml. This is a living document and contributions for clarifications or fixes and issue reports are highly welcome.
+
+## General
+Please start by reading:
+1. [CONTRIBUTING.md](../CONTRIBUTING.md).
+2. [C++ DEVELOPER_GUIDE.md](../cuML/DEVELOPER_GUIDE.md)
+3. [Python cuML README.md](README.md)
+
+## Thread safety
+Refer to the section on thread safety in [C++ DEVELOPER_GUIDE.md](../cuML/DEVELOPER_GUIDE.md#thread-safety)
+
+## Coding style
+1. [PEP8](https://www.python.org/dev/peps/pep-0008)
+2. [sklearn coding guidelines](https://scikit-learn.org/stable/developers/contributing.html#coding-guidelines)
+
+## Creating class for a new ML algo
+1. Make sure that this algo has been implemented in the C++ side. Refer to [C++ DEVELOPER_GUIDE.md](../cuML/DEVELOPER_GUIDE.md) for guidelines on developing in C++.
+2. Create a corresponding algoName.pyx file inside `python/cuml` folder.
```

Comment:
Perhaps mention here that the folder structure in the python/cuml should reflect the sklearn one, i.e. the algoName.pyx should be placed in a folder that mirrors the folder structure of sklearn.